scripts/backref_cost_interval_search_v1_paths.py

```
@@ -0,0 +1,177 @@
+        raise UnsafePath("run path must be a real directory, not a symlink")
+    canonical = os.path.realpath(candidate)
+    if canonical != candidate or os.path.dirname(canonical) != safe_root:
+        raise UnsafePath("canonical run path is not the exact safe-root child")
+    return {"home": home, "safe_root": safe_root, "run_directory": candidate,
+            "token": token, "validated": True}
+
+
+def generate() -> dict:
+    home = canonical_remote_home()
+    safe_root = ensure_safe_root(home)
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)